src/auth/index.js

```diff
@@ -36,14 +61,29 @@ const env = ({ authBucket, generateAuthBucket }, next) => (req, res) => {
     auth: oauth,
   }, {}, (err, response) => {
     if (response && response.permissions && response.permissions.includes(permission)) {
-      next(req, res);
+      resolve();
     } else {
-      res.status(403).json({
-        code: 403,
-        message: `The request is forbidden. ${bucket} did not allow access`,
-      });
+      reject({ code: 403, message: `The request is forbidden. ${bucket} did not allow access` });
     }
   });
+});
+
+const env = ({ authBucket, generateAuthBucket }, next) => (req, res) => {
+  let oauth = null;
+  return getAccessToken(req.get('Authorization'))
+    .then(retrieveUserInformation)
+    .then((info) => {
+      if (!req.body) req.body = {};
+      req.body = Object.assign(req.body, { gcframe: info });
```

Comment:
I don't think we should change the `body` of the request (e.g. requests don't always have a body?). I think we should add a new prop to `req` itself.

Reply:
Sure, would be easier actually to do it that way as well 😃